pkg/apiserver/profiling/model.go

```diff
@@ -25,12 +25,26 @@ import (
 // TaskState is used to represent the task/task group state.
 type TaskState int
 
+type GroupTaskState int
+
 // Built-in task state
+// State change flow:
+// For single task: TaskStateRunning --> TaskStateFailed (fail) | TaskStateFinish (success)
+// For group task:
+// - Old: TaskStateRunning      --> TaskStateFinish
+// - New: GroupTaskStateRunning --> GroupTaskStateAllFailed | GropuTaskStatePartialFailed | GroupTaskStateAllSuccess
+const (
+	TaskStateFailed  TaskState = iota // 0
+	TaskStateRunning                  // 1
+	TaskStateFinish                   // 2, for single task, it means success; for group task, it only means all child tasks are done, no matter they are success or failed, it will be refined by GroupTaskState
+)
+
 const (
-	TaskStateError TaskState = iota
-	TaskStateRunning
-	TaskStateFinish
-	TaskPartialFinish
+	GroupTaskStateUnknown       GroupTaskState = iota // 0
+	GroupTaskStateRunning                             // 1
```

Comment:
How about introducing `GroupTaskStateFinished = 2 // deprecated`? In this way, you can safely change GroupTask's state to the type `GroupTask` while maintaining compatibility with the old data.